atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/compact/BackgroundCompactor.java

```diff
@@ -284,12 +284,22 @@ private long getSleepTimeWhenCompactionHasNotRun() {
     }
 
     enum CompactionOutcome {
-        SUCCESS,
-        NOTHING_TO_COMPACT,
-        COMPACTED_BUT_NOT_REGISTERED,
-        UNABLE_TO_ACQUIRE_LOCKS,
-        FAILED_TO_COMPACT,
-        SHUTDOWN,
-        DISABLED
+        SUCCESS(0),
+        NOTHING_TO_COMPACT(1),
+        COMPACTED_BUT_NOT_REGISTERED(2),
+        UNABLE_TO_ACQUIRE_LOCKS(3),
+        FAILED_TO_COMPACT(4),
+        SHUTDOWN(5),
+        DISABLED(6);
+
+        private final int index;
+
+        CompactionOutcome(int index) {
+            this.index = index;
+        }
+
+        public int getIndex() {
+            return index;
+        }
```

Comment:
Used by production code for metrics. The tests seems enough to verify that no problem with functionality. Tho, this might effect old metrics if ordinals and indexes are different values? I am pretty sure the ordinal starts with 0 and goes consecutively if I am not missing weird jvm implementation etc.

From specification:

> Returns the ordinal of this enumeration constant (its position in its enum declaration, where the initial constant is assigned an ordinal of zero).